### PR TITLE
Docs: Improve release-process.md formatting

### PR DESCRIPTION
**Pull request by giulio92**
This `PR` will improve `release-process.md` formatting by:

1. Updating Markdown syntax to the latest one
2. Adding syntax highlighting to all code blocks
3. Aligning the text up to `80` column guideline (before it was following different guidelines, sometime `80`, sometime `90`, etc.)
4. Small grammar improvements here and there

### CODE REVIEW

**Comment by giulio92** [comment]
The Lint job of Travis CI just failed with this issue:
```
test/lint/lint-shell.sh:44: desriptor  ==> descriptor
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/lint-spelling.ignore-words.txt
This diff appears to have added new lines with trailing whitespace.
The following changes were suspected:
diff --git a/doc/release-process.md b/doc/release-process.md
@@ -13,3 +11,4 @@ Release Process
+* Update [`bips.md`](bips.md) to account for changes since the last release
+* Update version in [`configure.ac`](/configure.ac)
@@ -30,13 +27,25 @@ Release Process
+and `m_assumed_chain_state_size` with the current size plus some overhead
@@ -354,2 +401,5 @@ bitcoin.org (see below for bitcoin.org update instructions).
+        - Click "Request builds" for every new release on this branch
^---- failure generated from test/lint/lint-whitespace.sh
```

It's a false positive? This PR just changes a `.md` file, how could this be possibile?

**Comment by fanquake** [comment]
> This diff appears to have added new lines with trailing whitespace.

@giulio92 From the log. The problem is the trailing whitespace you've introduced.

File: doc/release-process.md [hunk 1]
```diff
@@ -1,168 +1,210 @@
...
 
 ## Branch updates
 
 ### Before every release candidate
-
-* Update translations (ping wumpus on IRC) see [translation_process.md](https://github.com/bitcoin/bitcoin/blob/master/doc/translation_process.md#synchronising-translations).
-* Update manpages, see [gen-manpages.sh](https://github.com/bitcoin/bitcoin/blob/master/contrib/devtools/README.md#gen-manpagessh).
-* Update release candidate version in `configure.ac` (`CLIENT_VERSION_RC`).
+* Update translations (ping [@wumpus](https://github.com/laanwj) on IRC) see [translation_process.md](./translation_process.md#synchronising-translations).
+* Update manpages, see [gen-manpages.sh](/contrib/devtools/README.md#gen-manpagessh).
```
Comment (hebasto) [inline doc/release-process.md]: back ticks around _gen-manpages.sh_ ?

File: doc/release-process.md [hunk 2]
```diff
@@ -1,168 +1,210 @@
-Release Process
-====================
+# Release Process
 
 ## Branch updates
 
 ### Before every release candidate
-
-* Update translations (ping wumpus on IRC) see [translation_process.md](https://github.com/bitcoin/bitcoin/blob/master/doc/translation_process.md#synchronising-translations).
```
Comment (hebasto) [inline doc/release-process.md]: back ticks around _translation_process.md_ ?

File: doc/release-process.md [hunk 3]
```diff
@@ -277,72 +329,67 @@ bitcoin-${VERSION}.tar.gz
...
-The `*-debug*` files generated by the gitian build contain debug symbols
-for troubleshooting by developers. It is assumed that anyone that is interested
-in debugging can run gitian to generate the files for themselves. To avoid
-end-user confusion about which file to pick, as well as save storage
-space *do not upload these to the bitcoin.org server, nor put them in the torrent*.
+The `*-debug*` files generated by the gitian build contain debug symbols for
+troubleshooting by developers. It is assumed that anyone that is interested in
+debugging can run gitian to generate the files for themselves. To avoid end-user
+confusion about which file to pick, as well as save storage space *do not upload
+these to the bitcoin.org server, nor put them in the torrent*.
```
Comment (hebasto) [inline doc/release-process.md]: Why this change?

File: doc/release-process.md [hunk 4]
```diff
@@ -368,19 +419,23 @@ bitcoin.org (see below for bitcoin.org update instructions).
 
   - Update title of #bitcoin on Freenode IRC
 
-  - Optionally twitter, reddit /r/Bitcoin, ... but this will usually sort out itself
+  - Optionally twitter, reddit /r/Bitcoin, ... but this will usually sort out
+  itself
```
Comment (hebasto) [inline doc/release-process.md]: Why this change?

**Comment by giulio92** [comment]
> > This diff appears to have added new lines with trailing whitespace.
> 
> @giulio92 From the log. The problem is the trailing whitespace you've introduced.

I understand.

Those trailing whitespace represent a line break in Markdown syntax ([reference](https://daringfireball.net/projects/markdown/syntax#p)) and I placed them to place the "**Note:**" section below the text it is referred to

**Before:**
Some text Note: some note

**After:**
Some text
**Note:** some note

In my opinion it looks better, should I remove the two trailing whitespaces?

File: doc/release-process.md (hunk 1 above)
Comment (giulio92) [reply doc/release-process.md]: ACK, applied change in 5a797b4

File: doc/release-process.md (hunk 2 above)
Comment (giulio92) [reply doc/release-process.md]: ACK, applied change in 5a797b4

**Comment by giulio92** [comment]
@hebasto regarding comments: [one](https://github.com/bitcoin/bitcoin/pull/17981#discussion_r369464795) and [two](https://github.com/bitcoin/bitcoin/pull/17981#discussion_r369465162) I aligned the text up to 80 column guideline to make it look more readable while editing the `.md` file

**Comment by hebasto** [comment]
> @hebasto regarding comments: [one](https://github.com/bitcoin/bitcoin/pull/17981#discussion_r369464795) and [two](https://github.com/bitcoin/bitcoin/pull/17981#discussion_r369465162) I aligned the text up to 80 column guideline to make it look more readable while editing the `.md` file

Our project [style guideline](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md) does not have such requirements. Though, it seems good be able not using horizontal scrolling on the GitHub while (re)viewing the code. It implies the maximum of line length about 118 characters.

**Comment by jonatack** [comment]
I don't think these style changes warrant the review effort for an internal process document.

**Comment by Empact** [comment]
Agree with @jonatack. To ease review, consider splitting it up into narrowly-constructed commits or separate pull requests. E.g. the syntax highlighting changes which would be trivial to review, particularly if you limited your changes to the bounds around the code.

**Comment by giulio92** [comment]
While already knowing that this document is something internal, I also think that, being Bitcoin an open source project, everyone should be able to read and understand how things works (even out of curiosity), and the changes of this PR goes in the direction of improving the formatting of the description of the release process so that everyone and not just who is the "release master" understands.

If we think that merging this changes is not useful I would prefer to close the PR now instead of starting a process of splitting the changes in separate commits and PRs

**Comment by jonatack** [comment]
@giulio92, FWIW here are several [in-depth articles](https://jonatack.github.io/articles) that you might find useful for contributing. This is only my opinion, but review time is valuable and scarce and more value is added by testing/reviewing important PRs and issues, or handling [good first issues](https://github.com/bitcoin/bitcoin/contribute) where help is actively requested. Cheers :)

**Comment by giulio92** [comment]
Closing this as it seems it does not bring any value

**Comment by Empact** [comment]
Thanks @giulio92, as Bitcoin is currently review-constrained, I encourage you to contribute but with attention to how you might reduce the review burden of your contributions, e.g. by making them more narrowly tailored.
